lib/devise/failure_app.rb

```diff
@@ -51,7 +51,18 @@ def http_auth
     end
 
     def recall
-      env["PATH_INFO"]  = attempted_path
+      config = Rails.application.config
+
+      if config.try(:relative_url_root)
+        base_path = Pathname.new(config.relative_url_root)
```

Comment:
Can we not use `Pathname` and call the methods in `File` or `Path` directly? I always find `Pathname` very error prone, specially if you forget to call `to_s`.